packages/indexer/src/handlers/events/price/handlePriceIndexing.event.handler.ts

```diff
@@ -0,0 +1,28 @@
+import stringify from "safe-stable-stringify"
+import { getBlockTimestamp } from "@/utils/rpc.helpers"
+import { getHostStateMachine } from "@/utils/substrate.helpers"
+import { wrap } from "@/utils/event.utils"
+import { TransferLog } from "@/configs/src/types/abi-interfaces/ERC6160Ext20Abi"
+import { PriceFeedsService } from "@/services/priceFeeds.service"
+
+/**
+ * Handle Price Indexing for all registered tokens on a chain when significant events occur
+ */
+export const handlePriceIndexing = wrap(async (transfer: TransferLog): Promise<void> => {
```

Comment:
I think you want to plug this into the block indexing hook for one of the chains(preferably hyperbridge),  it needs to run only in one place.